### ISSUE TITLE
goexports doesn't seem to play nicely with go modules

### ISSUE THREAD

**Issue by floren** [traefik/yaegi#427]
Several months ago, before my project switched over to Go modules, I did an experimental integration of yaegi. I used the interp package to instantiate a VM and run yaegi scripts. In order to make some of my project's internal packages available, I used `goexports` to generate symbol files, e.g.:
```
goexports project/pkg/base
```
This worked fine originally, but having switched over to Go modules and Go 1.13, I find that goexports no longer works properly. Here is a (slightly edited for clarity) error message:

```
	 goexports project/pkg/base
2019/10/24 09:20:34 type-checking package "project/pkg/base" failed (/home/john/go/src/project/pkg/base/base.go:26:2: could not import project/pkg/resourcemanager (type-checking package "project/pkg/resourcemanager" failed
[... elided ...]
could not import github.com/victorspringer/http-cache/adapter/memory (type-checking package "github.com/victorspringer/http-cache/adapter/memory" failed (/home/john/go/pkg/mod/github.com/victorspringer/http-cache@v0.0.0-20190721184638-fe78e97af707/adapter/memory/memory.go:33:2: could not import github.com/victorspringer/http-cache (cannot find package "github.com/victorspringer/http-cache" in any of:
	/usr/local/go/src/github.com/victorspringer/http-cache (from $GOROOT)
	/home/john/go/src/github.com/victorspringer/http-cache (from $GOPATH)))))))))))))
```

Basically, it seems as though once it got out of the packages in *my* repo and started to descend into external imports such as github.com/victorspringer/http-cache, it got lost.

I believe the problem may be related to this code in goexports.go:

```
func genContent(dest, pkgName, license string) ([]byte, error) {
	p, err := importer.ForCompiler(token.NewFileSet(), "source", nil).Import(pkgName)
```
The documentation for ForCompiler states:

> A lookup function must be provided for correct module-aware operation. Deprecated: If lookup is nil, for backwards-compatibility, the importer will attempt to resolve imports in the $GOPATH workspace.

I'm afraid I don't know enough about the Go build process, particularly about modules, to be able to suggest an actual solution.

**Comment by Equim-chan** [comment]
Maybe use [golang.org/x/tools/go/packages](https://godoc.org/golang.org/x/tools/go/packages) instead?

**Comment by mpl** [comment]
@floren the goexports (now `yaegi extract`) tool has been improved. Could you please retry and confirm we can close this issue?

**Comment by floren** [comment]
I finally got around to checking this and it appears to work fine now. Thank you very much!
